src/libexpr/eval.cc

```diff
@@ -264,30 +267,42 @@ static BoehmGCStackAllocator boehmGCStackAllocator;
  * However, the implementation is quite lean, and usually we don't have active
  * coroutines during evaluation, so this is acceptable.
  */
-void fixupBoehmStackPointer(void ** sp_ptr, void * pthread_id) {
+void fixupBoehmStackPointer(void ** sp_ptr, void * _pthread_id) {
     void *& sp = *sp_ptr;
+    auto pthread_id = reinterpret_cast<pthread_t>(_pthread_id);
     pthread_attr_t pattr;
     size_t osStackSize;
     void * osStackLow;
     void * osStackBase;
 
-    #ifdef __APPLE__
-    osStackSize = pthread_get_stacksize_np((pthread_t)pthread_id);
-    osStackLow = pthread_get_stackaddr_np((pthread_t)pthread_id);
-    #else
+#  ifdef __APPLE__
```

Comment:
I don't really like this indentation style because it looks very messy, and it's not what we do in the rest of the Nix repo. Can we indent `#ifdef`s the same as the rest of the code?

Reply:
Yeah the indentation thing is great for CPP alone, but less good for mixed regular code and CPP. Maybe I'll make this a separate issue.

Reply:
https://github.com/NixOS/nix/issues/10899 Made this issue.